src/modules/Game.class.js

```diff
@@ -6,63 +6,295 @@
+
+      const newRow = this.slide(row);
+
+      newRow.reverse();
+
+      if (newRow.toString() !== this.initialState[i].toString()) {
+        moved = true;
+      }
+      this.initialState[i] = newRow;
+
+      for (let c = 0; c < this.columns; c++) {
+        const tile = document.getElementById(i.toString() + '-' + c.toString());
+        const num = this.initialState[i][c];
+
+        this.updateTile(tile, num);
+      }
+    }
+
+    if (moved) {
+      this.addNewTile();
+    }
+    this.checkGameStatus();
+  }
+
+  moveUp() {
+    let moved = false;
+
+    for (let c = 0; c < this.columns; c++) {
+      const row = [
+        this.initialState[0][c],
+        this.initialState[1][c],
+        this.initialState[2][c],
+        this.initialState[3][c],
+      ];
+      const newRow = this.slide(row);
+
+      if (newRow.toString() !== row.toString()) {
+        moved = true;
+      }
+
+      for (let r = 0; r < this.rows; r++) {
+        this.initialState[r][c] = newRow[r];
+
+        const tile = document.getElementById(r.toString() + '-' + c.toString());
+        const num = this.initialState[r][c];
+
+        this.updateTile(tile, num);
+      }
+    }
+
+    if (moved) {
+      this.addNewTile();
+    }
+    this.checkGameStatus();
+  }
+
+  moveDown() {
+    let moved = false;
+
+    for (let c = 0; c < this.columns; c++) {
+      const row = [
+        this.initialState[0][c],
+        this.initialState[1][c],
+        this.initialState[2][c],
+        this.initialState[3][c],
+      ];
+
+      row.reverse();
+
+      const newRow = this.slide(row);
+
+      newRow.reverse();
+
+      if (newRow.toString() !== row.toString()) {
+        moved = true;
+      }
+
+      for (let r = 0; r < this.rows; r++) {
+        this.initialState[r][c] = newRow[r];
+
+        const tile = document.getElementById(r.toString() + '-' + c.toString());
+        const num = this.initialState[r][c];
+
+        this.updateTile(tile, num);
+      }
+    }
+
+    if (moved) {
+      this.addNewTile();
+    }
+    this.checkGameStatus();
+  }
+
+  addNewTile() {
+    const emptyTiles = [];
+
+    for (let r = 0; r < this.rows; r++) {
+      for (let c = 0; c < this.columns; c++) {
+        if (this.initialState[r][c] === 0) {
+          emptyTiles.push({ r, c });
+        }
+      }
+    }
+
+    if (emptyTiles.length > 0) {
+      const { r, c } =
+        emptyTiles[Math.floor(Math.random() * emptyTiles.length)];
+
+      this.initialState[r][c] = Math.random() < 0.9 ? 2 : 4;
+
+      const tile = document.getElementById(r.toString() + '-' + c.toString());
+
+      this.updateTile(tile, this.initialState[r][c]);
+    }
+  }
+
+  getScore() {
+    return this.score;
+  }
+
+  getState() {
+    return this.initialState;
+  }
+
+  checkGameStatus() {
+    for (let r = 0; r < this.rows; r++) {
+      for (let c = 0; c < this.columns; c++) {
+        if (this.initialState[r][c] === 2048) {
+          this.status = 'win';
+          alert('Você ganhou!');
```

Comment:
Using 'alert' for game notifications can be intrusive for the user experience. Consider implementing a custom modal or notification system within the game's UI.